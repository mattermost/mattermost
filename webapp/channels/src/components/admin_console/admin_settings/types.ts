// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {AdminConfig, ClientLicense} from '@mattermost/types/config';
import type {DeepPartial} from '@mattermost/types/utilities';

export type HandleSaveFunction = (config: DeepPartial<AdminConfig>, updateSettings: (id: string, value: unknown) => void) => void;
export type GetStateFromConfigFunction<State extends Record<string, any>> = (config: AdminConfig, license: ClientLicense) => State;
export type GetConfigFromStateFunction<State extends Record<string, any>> = (state: State) => DeepPartial<AdminConfig>;
