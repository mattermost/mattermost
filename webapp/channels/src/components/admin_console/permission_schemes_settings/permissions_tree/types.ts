// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {ClientLicense} from '@mattermost/types/config';

export type Permissions = Array<string | Group | Permission>;

export type Permission = {
    id: string;
    combined?: boolean;
    permissions: string[];
}
export type Group = {
    id: string;
    permissions: Array<Permission | string>;
    isVisible?: (license?: ClientLicense) => boolean;
}

export type AdditionalValues = Record<string, Record<string, any>>;
