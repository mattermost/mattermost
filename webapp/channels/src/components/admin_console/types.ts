// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {FormatXMLElementFn} from 'intl-messageformat';
import type {
    MessageDescriptor,
    PrimitiveType,
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    IntlShape,
} from 'react-intl';

import type {CloudState, Product} from '@mattermost/types/cloud';
import type {AdminConfig, ClientLicense} from '@mattermost/types/config';
import type {JobType} from '@mattermost/types/jobs';

import type Constants from 'utils/constants';

import type ValidationResult from './validation';

type Component = any

type AdminDefinitionConfigSchemaComponent = {
    id: string;
    component: Component;
}

export type ConsoleAccess = {read: {[key: string]: boolean}; write: {[key: string]: boolean}}

type Validator = (value: any) => ValidationResult

type AdminDefinitionSettingCustom = Omit<AdminDefinitionSettingBase, 'label'> & {
    type: 'custom';
    key: string;
    showTitle?: boolean;
    component: Component;
    label?: string;
}

type AdminDefinitionSettingBase = {
    key?: string;
    label: string | MessageDescriptor;
    label_values?: {[key: string]: any};
    help_text?: string | JSX.Element | MessageDescriptor;
    help_text_markdown?: boolean;
    help_text_values?: {[key: string]: any};
    disabled_help_text?: string | JSX.Element | MessageDescriptor;
    disabled_help_text_markdown?: boolean;
    disabled_help_text_values?: {[key: string]: any};
    onConfigLoad?: (configVal: any, config: Partial<AdminConfig>) => any;
    onConfigSave?: (displayVal: any, previousVal?: any) => any;
    isHidden?: Check;
    isDisabled?: Check;
}

export type AdminDefinitionSettingBanner = AdminDefinitionSettingBase & {
    type: 'banner';
    label_markdown?: boolean;
    banner_type: 'info' | 'warning';
}

type AdminDefinitionSettingRole = AdminDefinitionSettingBase & {
    type: 'roles';
    multiple?: boolean;
    no_result?: string | MessageDescriptor;
}

export type AdminDefinitionSettingInput = AdminDefinitionSettingBase & {
    type: 'text' | 'bool' | 'longtext' | 'number' | 'color';
    placeholder?: string | MessageDescriptor;
    placeholder_values?: {[key: string]: any};
    validate?: Validator;
    multiple?: boolean;
    setFromMetadataField?: string;
    dynamic_value?: (value: any, config: Partial<AdminConfig>, state: any) => string;
    max_length?: number;
    default?: string;
}

type AdminDefinitionSettingGenerated = AdminDefinitionSettingBase & {
    type: 'generated';
    placeholder?: string | MessageDescriptor;
    regenerate_help_text?: string;
    default?: string;
}

export type AdminDefinitionSettingDropdownOption = {
    value: string;
    display_name: string | MessageDescriptor;
    help_text?: string | MessageDescriptor;
    help_text_markdown?: boolean;
    help_text_values?: {[key: string]: any};
    isHidden?: Check;
}

type AdminDefinitionSettingDropdown = AdminDefinitionSettingBase & {
    type: 'dropdown';
    options: AdminDefinitionSettingDropdownOption[];
    isHelpHidden?: Check;
}

type AdminDefinitionSettingFileUpload = AdminDefinitionSettingBase & {
    type: 'fileupload';
    remove_help_text: string | MessageDescriptor;
    remove_button_text: string | MessageDescriptor;
    removing_text: string | MessageDescriptor;
    uploading_text: string | MessageDescriptor;
    fileType: string;
    upload_action: (file: File, success: (data: any) => void, error: (err: any) => void) => void;
    set_action?: () => void;
    setFromMetadataField?: string;
    remove_action: (success: (data: any) => void, error: (err: any) => void) => void;
}

type AdminDefinitionSettingJobsTable = AdminDefinitionSettingBase & {
    type: 'jobstable';
    job_type: JobType;
    render_job: Component;
};

type AdminDefinitionSettingLanguage = AdminDefinitionSettingBase & {
    type: 'language';
    multiple?: boolean;
    no_result?: string | MessageDescriptor;
}

type AdminDefinitionSettingButton = AdminDefinitionSettingBase & {
    type: 'button';
    action: (success: (data?: any) => void, error: (error: {message: string; detailed_error?: string}) => void, siteUrl: string) => void;
    loading?: string | MessageDescriptor;
    error_message: string | MessageDescriptor;
    success_message?: string | MessageDescriptor;
    sourceUrlKey?: string;
    skipSaveNeeded?: boolean;
}

type AdminDefinitionSettingUsername = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_USERNAME;
    placeholder: string;
    default?: string;
}

type MappingKeyTypes = 'enableTeamCreation' | 'editOthersPosts' | 'enableOnlyAdminIntegrations';

type AdminDefinitionSettingPermission = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_PERMISSION;
    permissions_mapping_name: MappingKeyTypes;
    key: string;
}

type AdminDefinitionSettingRadio = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_RADIO;
    options: AdminDefinitionSettingDropdownOption[];
    default?: string;
}

export type AdminDefinitionSetting = AdminDefinitionSettingCustom |
AdminDefinitionSettingInput | AdminDefinitionSettingGenerated |
AdminDefinitionSettingBanner | AdminDefinitionSettingDropdown |
AdminDefinitionSettingButton | AdminDefinitionSettingFileUpload |
AdminDefinitionSettingJobsTable | AdminDefinitionSettingLanguage |
AdminDefinitionSettingUsername | AdminDefinitionSettingPermission |
AdminDefinitionSettingRadio | AdminDefinitionSettingRole;

type AdminDefinitionConfigSchemaSettings = {
    id: string;
    name: string | MessageDescriptor;
    isHidden?: Check;
    onConfigLoad?: (config: Partial<AdminConfig>) => {[x: string]: string};
    onConfigSave?: (displayVal: any) => any;
    settings?: AdminDefinitionSetting[];
    sections?: AdminDefinitionConfigSchemaSection[];
    footer?: string | MessageDescriptor;
    header?: string | MessageDescriptor;
}

export type AdminDefinitionConfigSchemaSection = {
    key: string;
    title?: string;
    subtitle?: string;
    settings: AdminDefinitionSetting[];
    header?: string | MessageDescriptor;
    footer?: string | MessageDescriptor;
    component?: Component;
}

type RestrictedIndicatorType = {
    value: (cloud: CloudState) => JSX.Element;
    shouldDisplay: (license: ClientLicense, subscriptionProduct: Product|undefined) => boolean;
}

export type AdminDefinitionSubSectionSchema = AdminDefinitionConfigSchemaComponent | AdminDefinitionConfigSchemaSettings;

export type AdminDefinitionSubSection = {
    url: string;
    title?: string | MessageDescriptor;
    searchableStrings?: SearchableStrings;
    isHidden?: Check;
    isDiscovery?: boolean;
    isDisabled?: Check;
    schema: AdminDefinitionSubSectionSchema;
    restrictedIndicator?: RestrictedIndicatorType;
}

export type AdminDefinitionSection = {
    icon: JSX.Element;
    sectionTitle: string | MessageDescriptor;
    isHidden: Check;
    id?: string;
    subsections: {[key: string]: AdminDefinitionSubSection};
}

/** From {@link IntlShape.formatMessage}. Cannot discriminate overloaded method signature. */
declare function formatMessageBasic(descriptor: MessageDescriptor, values?: Record<string, PrimitiveType | FormatXMLElementFn<string, string>>): string;

export type SearchableStrings = Array<string | MessageDescriptor | Parameters<typeof formatMessageBasic>>;

export type AdminDefinition = {[key: string]: AdminDefinitionSection}

export type Check = boolean | ((config: Partial<AdminConfig>, state: any, license?: ClientLicense, enterpriseReady?: boolean, consoleAccess?: ConsoleAccess, cloud?: CloudState, isSystemAdmin?: boolean) => boolean)
