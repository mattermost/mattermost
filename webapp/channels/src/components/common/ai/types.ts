// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// Re-export the AIAgent type from mattermost-redux
export type {AIAgent as BridgeBotInfo} from 'mattermost-redux/actions/ai';

