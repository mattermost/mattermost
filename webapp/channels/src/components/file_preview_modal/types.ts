// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {FileInfo} from '@mattermost/types/files';

export type LinkInfo = {
    has_preview_image: boolean;
    link: string;
    extension: string;
    name: string;
}

export function isFileInfo(info: FileInfo | LinkInfo): info is FileInfo {
    return Boolean((info as FileInfo).id);
}

export function isLinkInfo(info: FileInfo | LinkInfo): info is LinkInfo {
    return Boolean((info as LinkInfo).link) && !isFileInfo(info);
}
