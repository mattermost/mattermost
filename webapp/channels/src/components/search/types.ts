// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import type {Action} from 'redux';

import type {UserAutocomplete} from '@mattermost/types/autocomplete';
import type {Channel} from '@mattermost/types/channels';
import type {ServerError} from '@mattermost/types/errors';

import type {SearchType} from 'types/store/rhs';

export type SearchFilterType = 'all' | 'documents' | 'spreadsheets' | 'presentations' | 'code' | 'images' | 'audio' | 'video';

export type OwnProps = {
    isSideBarRight?: boolean;
    isSideBarRightOpen?: boolean;
    hideSearchBar?: boolean;
    enableFindShortcut?: boolean;
    channelDisplayName?: string;
    getFocus?: (searchBarFocus: () => void) => void;
    children?: React.ReactNode;
}

export type StateProps = {
    isRhsExpanded: boolean;
    isRhsOpen: boolean;
    isSearchingTerm: boolean;
    searchTerms: string;
    searchTeam: string;
    searchType: SearchType;
    searchVisible: boolean;
    hideMobileSearchBarInRHS: boolean;
    isMentionSearch: boolean;
    isFlaggedPosts: boolean;
    isPinnedPosts: boolean;
    isChannelFiles: boolean;
    currentChannel?: Channel;
    isMobileView: boolean;
    crossTeamSearchEnabled: boolean;
}

export type DispatchProps = {
    actions: {
        updateSearchTerms: (term: string) => Action;
        updateSearchTeam: (teamId: string|null) => Action;
        updateSearchTermsForShortcut: () => void;
        updateSearchType: (searchType: string) => Action;
        showSearchResults: (isMentionSearch: boolean) => unknown;
        showChannelFiles: (channelId: string) => void;
        showMentions: () => void;
        showFlaggedPosts: () => void;
        setRhsExpanded: (expanded: boolean) => Action;
        closeRightHandSide: () => void;
        autocompleteChannelsForSearch: (term: string, success?: (channels: Channel[]) => void, error?: (err: ServerError) => void) => void;
        autocompleteUsersInTeam: (username: string) => Promise<UserAutocomplete>;
        updateRhsState: (rhsState: string) => void;
        getMorePostsForSearch: () => void;
        openRHSSearch: () => void;
        getMoreFilesForSearch: () => void;
        filterFilesSearchByExt: (extensions: string[]) => void;
    };
}

export type Props = StateProps & DispatchProps & OwnProps;
