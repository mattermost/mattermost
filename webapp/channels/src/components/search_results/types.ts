// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import type {IntlShape} from 'react-intl';

import type {FileInfo} from '@mattermost/types/files';
import type {Post} from '@mattermost/types/posts';

import type {SearchFilterType} from 'components/search/types';

import type {SearchType} from 'types/store/rhs';

export type OwnProps = {
    [key: string]: any;
    isSideBarExpanded: boolean;
    isMentionSearch: boolean;
    isFlaggedPosts: boolean;
    isPinnedPosts: boolean;
    updateSearchTerms: (terms: string) => void;
    getMorePostsForSearch: () => void;
    getMoreFilesForSearch: () => void;
    shrink: () => void;
    isCard?: boolean;
    isOpened?: boolean;
    channelDisplayName?: string;
    children?: React.ReactNode;
    searchType: SearchType;
    setSearchType: (searchType: SearchType) => void;
    searchFilterType: SearchFilterType;
    setSearchFilterType: (filterType: SearchFilterType) => void;
    updateSearchTeam: (teamId: string) => void;
    crossTeamSearchEnabled: boolean;
};

export type StateProps = {
    results: Post[];
    fileResults: FileInfo[];
    matches: Record<string, string[]>;
    searchTerms: string;
    searchSelectedType: string;
    isSearchingTerm: boolean;
    isSearchingFlaggedPost: boolean;
    isSearchingPinnedPost: boolean;
    isSearchGettingMore: boolean;
    isSearchAtEnd: boolean;
    isSearchFilesAtEnd: boolean;
};

export type IntlProps = {
    intl: IntlShape;
};

export type Props = OwnProps & StateProps & IntlProps;
