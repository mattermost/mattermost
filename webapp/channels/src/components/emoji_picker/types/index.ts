// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {MessageDescriptor} from 'react-intl';

import type {EmojiCategory, Emoji, SystemEmoji, CustomEmoji} from '@mattermost/types/emojis';

import type {
    CATEGORY_HEADER_ROW,
    EMOJIS_ROW,
} from 'components/emoji_picker/constants';

export type Category = {
    name: EmojiCategory;
    emojiIds?: string[];
    label: MessageDescriptor;
    iconClassName: string;
};

export type Categories = Record<EmojiCategory, Category>;

export type CategoryOrEmojiRow = CategoryHeaderRow | EmojiRow;

export type CategoryHeaderRow = {
    index: number;
    type: typeof CATEGORY_HEADER_ROW;
    items: Array<{
        categoryIndex: number;
        categoryName: EmojiCategory;
        emojiIndex: -1;
        emojiId: '';
        item: undefined;
    }>;
}

export type EmojiRow = {
    index: number;
    type: typeof EMOJIS_ROW;
    items: Array<{
        categoryIndex: number;
        categoryName: EmojiCategory;
        emojiIndex: number;
        emojiId: CustomEmoji['id'] | SystemEmoji['unified'];
        item: Emoji;
    }>;
}

export type EmojiCursor = {
    rowIndex: number;
    emojiId: CustomEmoji['id'] | SystemEmoji['unified'];
    emoji: Emoji | undefined;
};

export type EmojiPosition = {
    rowIndex: number;
    emojiId: CustomEmoji['id'] | SystemEmoji['unified'];
    categoryName: EmojiCategory;
}

export enum NavigationDirection {
    NextEmoji = 'next',
    PreviousEmoji = 'previous',
    NextEmojiRow = 'nextRow',
    PreviousEmojiRow = 'previousRow',
}
