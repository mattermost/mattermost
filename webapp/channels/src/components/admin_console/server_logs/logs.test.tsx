// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {LogLevelEnum} from '@mattermost/types/admin';

import {renderWithContext, userEvent, screen, waitFor} from 'tests/react_testing_utils';

import Logs from './logs';

describe('components/admin_console/server_logs/Logs', () => {
    // Log dataset
    const logs = [{
        caller: 'caller 1',
        job_id: 'job_id 1',
        level: LogLevelEnum.INFO,
        msg: 'msg 1',
        timestamp: 'timestamp 1',
        worker: 'worker 1',
        whatever: 'whatever 1',
    }, {
        caller: 'caller 2',
        job_id: 'job_id 2',
        level: LogLevelEnum.INFO,
        msg: 'msg 2',
        timestamp: 'timestamp 2',
        worker: 'worker 2',
        whatever: 'whatever 2',
    }, {
        caller: 'filtered',
        job_id: 'filtered',
        level: LogLevelEnum.INFO,
        msg: 'filtered message',
        timestamp: 'filtered',
        worker: 'filtered',
        whatever: 'filtered',
    }];
    let container: HTMLElement;

    beforeEach(async () => {
        // Mount server log screen
        container = renderWithContext(
            <Logs
                logs={logs}
                plainLogs={[]}
                isPlainLogs={false}
                actions={{getLogs: jest.fn(), getPlainLogs: jest.fn()}}
            />,
        ).container;

        // Wait for the logs to be displayed
        await waitFor(() => {
            expect(screen.queryByText('Loading')).not.toBeInTheDocument();
            expect(screen.queryByText('msg 1')).toBeInTheDocument();
            expect(screen.queryByText('msg 2')).toBeInTheDocument();
            expect(screen.queryByText('filtered message')).toBeInTheDocument();
        });
    });

    test('should display the logs correctly after loading', () => {
        expect(container).toMatchSnapshot();
    });

    test.each(['caller', 'msg', 'worker', 'job_id', 'whatever'])('should search input be performed on %s attribute',
        async (searchString: string) => {
            const searchInput = screen.getByTestId('searchInput');
            userEvent.type(searchInput, searchString);

            await waitFor(() => {
                expect(screen.queryByText('msg 1')).toBeInTheDocument();
                expect(screen.queryByText('msg 2')).toBeInTheDocument();
                expect(screen.queryByText('filtered message')).not.toBeInTheDocument();
            });
        });

    test.each(['level', 'timestamp'])('should search input not be performed on %s attribute',
        async (searchString: string) => {
            const searchInput = screen.getByTestId('searchInput');
            userEvent.type(searchInput, searchString);

            await waitFor(() => {
                expect(screen.queryByText('msg 1')).not.toBeInTheDocument();
                expect(screen.queryByText('msg 2')).not.toBeInTheDocument();
                expect(screen.queryByText('filtered message')).not.toBeInTheDocument();
            });
        });
});
