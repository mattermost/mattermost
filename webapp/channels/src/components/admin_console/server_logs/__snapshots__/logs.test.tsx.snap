// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/server_logs/Logs should display the logs correctly after loading 1`] = `
<div>
  <div
    class="wrapper--admin"
  >
    <div
      class="admin-console__header"
    >
      Server Logs
    </div>
    <div
      class="admin-console__wrapper"
    >
      <div
        class="admin-logs-content admin-console__content"
      >
        <div
          class="logs-banner"
        >
          <div
            class="banner"
          >
            <div
              class="banner__content"
            >
              To look up users by User ID or Token ID, go to User Management &gt; Users and paste the ID into the search filter.
            </div>
          </div>
          <div
            class="banner-buttons"
          >
            <div
              aria-labelledby="admin.logs.LogFormat.legend"
              class="banner-buttons__log-format"
              id="admin.logs.LogFormat"
              role="radiogroup"
            >
              <span
                id="admin.logs.LogFormat.legend"
              >
                Log Format:
              </span>
              <label>
                <input
                  checked=""
                  id="admin.logs.LogFormat.json"
                  name="log-format"
                  type="radio"
                  value="json"
                />
                JSON
              </label>
              <label>
                <input
                  id="admin.logs.LogFormat.plain"
                  name="log-format"
                  type="radio"
                  value="plain"
                />
                Plain text
              </label>
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Reload Logs
            </button>
            <a
              class="btn btn-primary"
              href="/api/v4/logs/download"
              location="download_logs"
              rel="noopener noreferrer"
              target="_blank"
            >
              Download Logs
            </a>
          </div>
        </div>
        <div
          class="LogTable"
        >
          <div
            class="DataGrid"
          >
            <div
              class="DataGrid_search"
            >
              <div
                class="DataGrid_searchBar"
              >
                <span
                  aria-hidden="true"
                  class="DataGrid_searchIcon"
                >
                  <i
                    class="fa fa-search"
                    title="Search Icon"
                  />
                </span>
                <input
                  data-testid="searchInput"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <i
                  class="DataGrid_clearButton fa fa-times-circle hidden"
                  data-testid="clear-search"
                />
              </div>
              <div
                class="Filter"
              >
                <button
                  class="Filter_button"
                  type="button"
                >
                  <i
                    class="Icon icon-filter-variant"
                  />
                  Filters
                </button>
                <div
                  class="Filter_content"
                >
                  <div
                    class="Filter_header"
                  >
                    <div
                      class="Filter_title"
                    >
                      Filter by
                    </div>
                    <a
                      class="Filter_reset"
                    >
                      Reset filters
                    </a>
                  </div>
                  <hr />
                  <div
                    class="Filter_lists"
                  >
                    <div
                      class="FilterList"
                    >
                      <div
                        class="FilterList_name"
                      >
                        Levels
                      </div>
                      <div
                        class="FilterList_item"
                      >
                        <div
                          class="FilterList_checkbox"
                        >
                          <label>
                            <input
                              checked=""
                              id="all"
                              name="all"
                              type="checkbox"
                            />
                            All levels
                          </label>
                        </div>
                      </div>
                      <div
                        class="FilterList_item"
                      >
                        <div
                          class="FilterList_checkbox"
                        >
                          <label>
                            <input
                              id="error"
                              name="error"
                              type="checkbox"
                            />
                            Error
                          </label>
                        </div>
                      </div>
                      <div
                        class="FilterList_item"
                      >
                        <div
                          class="FilterList_checkbox"
                        >
                          <label>
                            <input
                              id="info"
                              name="info"
                              type="checkbox"
                            />
                            Info
                          </label>
                        </div>
                      </div>
                      <div
                        class="FilterList_item"
                      >
                        <div
                          class="FilterList_checkbox"
                        >
                          <label>
                            <input
                              id="debug"
                              name="debug"
                              type="checkbox"
                            />
                            Debug
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                  <button
                    class="btn btn-primary btn-sm Filter_apply"
                    disabled=""
                    type="button"
                  >
                    Apply
                  </button>
                </div>
              </div>
              <button
                class="btn btn-tertiary btn-sm ml-2"
              >
                Show last 0 errors
              </button>
            </div>
            <div
              class="DataGrid_header"
            >
              <div
                class="DataGrid_cell"
                style="flex-grow: 1.5;"
              >
                <div
                  class="timestamp"
                >
                  Timestamp
                  <svg
                    fill="currentColor"
                    height="18"
                    version="1.1"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13,20H11V8L5.5,13.5L4.08,12.08L12,4.16L19.92,12.08L18.5,13.5L13,8V20Z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="DataGrid_cell"
                style="flex-grow: 0.5;"
              >
                Level
              </div>
              <div
                class="DataGrid_cell"
                style="flex-grow: 2.5;"
              >
                Message
              </div>
              <div
                class="DataGrid_cell"
                style="flex-grow: 1.5;"
              >
                Caller
              </div>
              <div
                class="DataGrid_cell"
                style="flex-grow: 1;"
              >
                Options
              </div>
            </div>
            <div
              class="DataGrid_rows"
              style="min-height: 120px;"
            >
              <div
                class="DataGrid_row"
              >
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-name overflow--ellipsis row-content"
                    data-testid="timestamp"
                  >
                    <span
                      class="group-description row-content"
                    >
                      timestamp 1
                    </span>
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 0.5; text-align: left;"
                >
                  <span
                    class="group-description adjusted row-content"
                  >
                    info
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 2.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                    title="msg 1"
                  >
                    msg 1
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                  >
                    caller 1
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1; text-align: left;"
                >
                  <button
                    class="btn btn-secondary btn-sm"
                    type="submit"
                  >
                    Full Log event
                  </button>
                </div>
              </div>
              <div
                class="DataGrid_row"
              >
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-name overflow--ellipsis row-content"
                    data-testid="timestamp"
                  >
                    <span
                      class="group-description row-content"
                    >
                      timestamp 2
                    </span>
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 0.5; text-align: left;"
                >
                  <span
                    class="group-description adjusted row-content"
                  >
                    info
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 2.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                    title="msg 2"
                  >
                    msg 2
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                  >
                    caller 2
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1; text-align: left;"
                >
                  <button
                    class="btn btn-secondary btn-sm"
                    type="submit"
                  >
                    Full Log event
                  </button>
                </div>
              </div>
              <div
                class="DataGrid_row"
              >
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-name overflow--ellipsis row-content"
                    data-testid="timestamp"
                  >
                    <span
                      class="group-description row-content"
                    >
                      filtered
                    </span>
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 0.5; text-align: left;"
                >
                  <span
                    class="group-description adjusted row-content"
                  >
                    info
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 2.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                    title="filtered message"
                  >
                    filtered message
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1.5; text-align: left;"
                >
                  <span
                    class="group-description row-content"
                  >
                    filtered
                  </span>
                </div>
                <div
                  class="DataGrid_cell"
                  style="flex-grow: 1; text-align: left;"
                >
                  <button
                    class="btn btn-secondary btn-sm"
                    type="submit"
                  >
                    Full Log event
                  </button>
                </div>
              </div>
            </div>
            <div
              class="DataGrid_footer"
            >
              <div
                class="DataGrid_cell"
              >
                1 - 3 of 3
                <button
                  class="btn btn-quaternary btn-icon btn-sm ml-2 prev disabled"
                  disabled=""
                  type="button"
                >
                  <i
                    class="icon icon-chevron-left"
                    title="Previous Icon"
                  />
                </button>
                <button
                  class="btn btn-quaternary btn-icon btn-sm next disabled"
                  disabled=""
                  type="button"
                >
                  <i
                    class="icon icon-chevron-right"
                    title="Next Icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
